cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/FileSetTest.java

```diff
@@ -227,6 +234,24 @@ public void testExternalNonExistentPath() throws IOException, DatasetManagementE
                                      .build());
   }
 
+
+  @Test(expected = IOException.class)
```

Comment:
In fact, strictly speaking, you don't need to create a fileset to create a directory right? You can just create a temporary directory, assert that it exists, then create a fileset at that location and fail?